### PR TITLE
Fix placeholder accessor wrong checking

### PR DESCRIPTION
**Pull request by aleksmesh**
Moved placeholder accessor constructor checking from sycl::handler command.
Cherry-picked PR #669 to avoid CI failures.

To review changes follow this [link](https://github.com/KhronosGroup/SYCL-CTS/pull/683/commits/a9915ad89d9cd4efd808751c5b538be7fde370c8..HEAD).

### CODE REVIEW

**Comment by keryell** [comment]
I can wait for the other PRs to land first.

**Comment by bader** [comment]
> I can wait for the other PRs to land first.

@keryell, I would appreciate if you could review [the patch](https://github.com/KhronosGroup/SYCL-CTS/pull/683/commits/a9915ad89d9cd4efd808751c5b538be7fde370c8..c081921777eb7915d5a30f078ccfdfcfb780e659) before #669 is landed.

File: tests/accessor/accessor_common.h
Comment (keryell) [inline tests/accessor/accessor_common.h:421]: Why is there a difference for iterators between host and device?

**Review by ProGTX — APPROVED** [review]

File: tests/accessor/accessor_common.h
Comment (keryell) [reply tests/accessor/accessor_common.h:427]: Could you summarize this in some comments around the code? We might forget about the rationale... :-)

Comment (aleksmesh) [reply tests/accessor/accessor_common.h:433]: Comments added.

**Comment by bader** [comment]
@aleksmesh, could you take a look at new conflicts, please?

**Comment by aleksmesh** [comment]
@bader Conflicts are merged.

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**
